radixdlt/src/integration/java/com/radixdlt/consensus/deterministic/ControlledBFTNetwork.java

```diff
@@ -56,6 +61,14 @@
 					key -> new LinkedList<>()
 				)
 			);
+
+		this.numberOfProposalsByNodes = new HashMap<>();
```

Comment:
I think these proposals should be computed outside of the network as it is a property that can be measured from the outside

Reply:
more details plz! How? Do you mean from `	private final ImmutableList<ControlledBFTNode> nodes;` within `BFTDeterministicTest` and then `SystemCounters systemCounters` from within `ControlledBFTNode`? If you mean the `SystemCounters` which CounterType? The `CONSENSUS_VIEW` ? Or did you mean in any other way?

Reply:
Or did you mean not at all and then change the `processNextMsg` to return messages and check for (type cast) proposals?